### DISCUSSION TITLE
Pairing problem

### DISCUSSION THREAD

**Discussion by narf007** [HomeSpan/HomeSpan#44] [Q&A]
Hello,

I'm just starting with "HomeSpan" library and having some problems during the pairing phase. I'm using the AZDelivery ESP32 D1 Mini NodeMCU. For my first tests, I tried the application "05-WorkingLED" and the Home application on my iPhone. Everything was OK. Then  I tried to use "mrstegeman / hap-controller-node" library to build my own contrôler but I'd never managed to complete the pairing. 

As your library is working well with the Home application on my iPhone, I started to look if anything was wrong with my controler. But my controler is also working great with other Homekit devices.  

I looked for clues in the documentation on wich you did a great job but I wasn’t able to find any usefull information to solve my problem. 

I set the log level 2 to get more information. I got  "*** ERROR: Pair Method not set to 0" . I don't have enough knowledge in the Homekit protocol to understand on wich side the problem is. Could you help correcting my issue ?  

Some more details from the serial monitor  :
```
<<<<<<<<< 192.168.0.51 <<<<<<<<<
POST /pair-setup HTTP/1.1
Content-Type: application/pairing+tlv8
Content-Length: 6
------------ END BODY! ------------
STATE(1) 01
METHOD(1) 01
------------ END TLVS! ------------
In Pair Setup...Found <M1>.  Expected <M1>

*** ERROR: Pair Method not set to 0


>>>>>>>>>> 192.168.0.51 >>>>>>>>>>
HTTP/1.1 200 OK
Content-Type: application/pairing+tlv8
Content-Length: 6

STATE(1) 02
ERROR(1) 06
------------ SENT! --------------

```

**Comment by HomeSpan** [comment]
It appears that your Controller is trying to connect to HomeSpan using Apple's Commercial protocol (MiFi) instead of the non-commercial HAP protocol.  As a non-commercial hobby product, HomeSpan is not certified by Apple (hence the warning message when you pair with the Home App).  HomeSpan broadcasts its non-commercial status as part of the pairing process, but it seems your Controller is either not recognizing HomeSpan's non-commercial status, or does not have the ability to connect to non-commercial devices.

From the log file, we see that STATE=1, which correctly signals HomeSpan to start the pairing process, but the METHOD=1.  This indicates commercial pairing using MiFi certification procols.   HomeSpan is expecting METHOD=0, which indicates non-commercial pairing.

To test this theory further, I removed the error check in HomeSpan and changed it to advertise itself as a commercial device instead of a non-commercial device.  When trying to pair with the Home App, I did not receive a warning message about the device being uncertified (since the Home App thought it was a commercial device), and the log file now showed METHOD=1, instead of METHOD=0 (as it normally does).  But since HomeSpan is not actually a commercial device and does not have a MiFi certification code, the pairing failed a few steps later as expected.

Have you used your Controller with any other non-commercial products (such as HomeBridge?), or only commercial HomeKit devices?

**Comment by narf007** [comment]
Thanks for your quick and documented answer and for your testing. I'm quite surprised. To implement the controller, I use an open source library which is certainly not an FMI solution (I think so). I'll check if there is any option to fix one or the other advertising method. Yes, I have used my controller with other non commercial products (other homekit open source solutions with some ESP32 and ESP8266), but it is the first time I  encounter such issue. I'll let you know what I can find out.
